pkg/universal_sigver/client_test.go

```diff
@@ -0,0 +1,142 @@
+package universal_sigver
+
+import (
+	"context"
+	"math/big"
+	"testing"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/crypto"
+	signer_pkg "github.com/layer-3/clearsync/pkg/signer"
+	"github.com/layer-3/clearsync/pkg/smart_wallet"
+	"github.com/layer-3/clearsync/pkg/userop"
+	"github.com/layer-3/clearsync/pkg/userop/testing/local_blockchain"
+	"github.com/shopspring/decimal"
+	"github.com/stretchr/testify/require"
+)
+
+func TestVerify(t *testing.T) {
+	ctx := context.Background()
+
+	// 1. Setup node, contracts, signer
+	node := local_blockchain.NewEthNode(ctx, t)
+	contracts := local_blockchain.SetupContracts(ctx, t, node)
+	eoa, err := local_blockchain.NewAccountWithBalance(ctx, decimal.NewFromFloat(2e18).BigInt(), node)
+	require.NoError(t, err)
+	signer := signer_pkg.NewLocalSigner(eoa.PrivateKey)
+
+	// 2. Setup smart wallet config, sigver and sign message
+	swCfg := smart_wallet.Config{
+		Type:           &smart_wallet.KernelType,
+		Factory:        contracts.Factory,
+		Logic:          contracts.Logic,
+		ECDSAValidator: contracts.ECDSAValidator,
+	}
+	sigver := NewUniversalSigVer(node.LocalURL.String(), &swCfg, &contracts.EntryPoint)
+	msg := []byte("hello")
+	sig, err := signer_pkg.SignEthMessage(signer, msg)
+	require.NoError(t, err)
+	msgHash := common.BytesToHash(signer_pkg.ComputeEthereumSignedMessageHash(msg))
+	rawSig := sig.Raw()
+
+	t.Run("Successfuly verify ECDSA signature", func(t *testing.T) {
+		ok, err := sigver.Verify(ctx, signer.CommonAddress(), msgHash, rawSig)
+		require.NoError(t, err)
+		require.True(t, ok)
+	})
+
+	t.Run("Successfuly verify ERC-1271 signature", func(t *testing.T) {
+		index := decimal.NewFromInt(0)
```

Comment:
Consider using `decimal.Zero` as it serves as a shorthand for `decimal.NewFromInt(0)`, streamlining the code.

Reply:
03b2806414bea4f017c757d5a577c632ea17f7b9